source/iterator/position.go

```diff
@@ -19,33 +19,55 @@ import (
 	"fmt"
 
 	sdk "github.com/conduitio/conduit-connector-sdk"
+	"vitess.io/vitess/go/vt/proto/binlogdata"
 )
 
 // Mode defines an iterator mode.
 type Mode string
 
 const (
-	// IteratorTypeSnapshot represents a snapshot iterator type.
+	// ModeSnapshot represents a snapshot iterator mode.
 	ModeSnapshot = "snapshot"
-	// IteratorTypeCDC represents a CDC iterator type.
+	// ModeCDC represents a CDC iterator mode.
 	ModeCDC = "cdc"
 )
 
 // Position is a combined iterator's position.
 type Position struct {
 	Mode Mode `json:"mode"`
+	// Keyspace holds a name of a VTGate keyspace that the connector interact with.
+	// Mode: snapshot, cdc.
+	Keyspace string `json:"keyspace"`
 	// LastProcessedElementValue is a value of the element
 	// at which the iterator stopped reading rows.
 	// The iterator will continue reading from the element if it's not empty.
-	// IteratorType: snapshot, cdc.
+	// Mode: snapshot, cdc.
 	LastProcessedElementValue any `json:"last_processed_element_value,omitempty"`
-	// Gtid specifies a gtid to start with.
-	// IteratorType: cdc.
+	// Gtid is the most recent gtid to start with.
+	// Mode: cdc.
 	Gtid string `json:"gtid,omitempty"`
+	// ShardGtids holds a list of shards which the connector will read events from.
+	// Mode: cdc.
+	ShardGtids []*binlogdata.ShardGtid `json:"shard_gtids,omitempty"`
 }
 
-// marshalPosition marshals the underlying position into a sdk.Position as JSON bytes.
-func (p *Position) marshalSDKPosition() (sdk.Position, error) {
+// GetBinlogShardGtids returns Position's ShardGtids and
+// sets the ShardGtids's Gtid field to the Position's Gtid which is the most recent Gtid.
+func (p *Position) GetBinlogShardGtids() []*binlogdata.ShardGtid {
+	if len(p.ShardGtids) == 0 {
+		return nil
```

Comment:
if ShardGtids = nil it is mean that cdc not started or cdc data is empty, am I right?

Reply:
Yeah, it means that the CDC iterator is not started/initialized

Reply:
If the `ShardGtids` field is empty or nil, we will get an error at this line https://github.com/conduitio-labs/conduit-connector-vitess/blob/c3aa1b78a1c16618e5a36b9b4e82694c8f280885/source/iterator/cdc.go#L165 and the CDC iterator won't start